pex/cli/commands/interpreter.py

```diff
@@ -0,0 +1,150 @@
+            parser,
+            description="Interact with local interpreters via the following subcommands.",
+        )
+        with subcommands.parser(
+            name="inspect",
+            help="Inspect local interpreters",
+            func=cls._inspect,
+            include_verbosity=False,
+        ) as inspect_parser:
+            cls._add_inspect_arguments(inspect_parser)
+
+    def _inspect(self):
+        # type: () -> Result
+
+        interpreter_configuration = target_options.configure_interpreters(self.options)
+        interpreters = interpreter_configuration.resolve_interpreters()
+        if self.options.all:
+            python_path = (
+                interpreter_configuration.python_path.split(":")
+                if interpreter_configuration.python_path
+                else None
+            )
+            interpreters.update(PythonInterpreter.all(paths=python_path))
+        if not interpreters:
+            interpreters.add(PythonInterpreter.get())
+
+        verbose = self.options.verbose or self.options.tags or self.options.markers
+        if self.options.indent and not verbose:
+            logger.warning(
+                "Ignoring --indent={} since --verbose mode is not enabled.".format(
+                    self.options.indent
+                )
+            )
+        with self.output(self.options) as out:
+            try:
+                for interpreter in interpreters:
+                    if verbose:
+                        interpreter_info = {"path": interpreter.binary}  # type: Dict[str, Any]
+                        if self.options.verbose:
+                            interpreter_info.update(
+                                version=interpreter.identity.version_str,
+                                requirement=str(interpreter.identity.requirement),
+                                platform=str(interpreter.platform),
+                                venv=interpreter.is_venv,
+                            )
+                            if interpreter.is_venv:
+                                interpreter_info[
+                                    "base_interpreter"
+                                ] = interpreter.resolve_base_interpreter().binary
+                        if self.options.tags:
+                            interpreter_info["compatible_tags"] = [
+                                str(tag) for tag in interpreter.identity.supported_tags
+                            ]
+                        if self.options.markers:
+                            interpreter_info[
+                                "marker_environment"
+                            ] = interpreter.identity.env_markers.as_dict()
+                        self.dump_json(self.options, interpreter_info, out)
+                    else:
+                        out.write(interpreter.binary)
+                    out.write("\n")
+            except UnsatisfiableInterpreterConstraintsError as e:
+                return Error(str(e))
```

Comment:
Where does the error come from? `interpreter.resolve_base_interpreter()`? It looks everything else is only creating a dictionary and accessing attr fields, which I wouldn't expect to be fallible.

If possible, it would be helpful to make this `try except` block tighter.